sql/src/test/java/io/crate/protocols/postgres/PostgresWireProtocolTest.java

```diff
@@ -384,4 +393,74 @@ public void testSessionCloseOnTerminationMessage() throws Exception {
 
         verify(session, times(1)).close();
     }
+
+    @Test
+    public void testHandleMultipleSimpleQueries() {
+        SQLOperations sqlOperations = Mockito.mock(SQLOperations.class);
+        Session session = mock(Session.class);
+        when(sqlOperations.createSession(any(String.class), any(User.class))).thenReturn(session);
+        Session.DescribeResult describeResult = mock(Session.DescribeResult.class);
+        when(describeResult.getFields()).thenReturn(null);
+        when(session.describe(anyChar(), anyString())).thenReturn(describeResult);
+        when(session.sync()).thenReturn(CompletableFuture.completedFuture(false));
+
+        PostgresWireProtocol ctx =
+            new PostgresWireProtocol(
+                sqlOperations,
+                new AlwaysOKNullAuthentication(),
+                null);
+        EmbeddedChannel channel = new EmbeddedChannel(ctx.decoder, ctx.handler);
+
+        sendStartupMessage(channel);
+        readAuthenticationOK(channel);
+        skipParameterMessages(channel);
+        readReadyForQueryMessage(channel);
+
+        ByteBuf query = Unpooled.buffer();
+        try {
+            Messages.writeCString(query, "set search_path to 'hoschi';select 42;".getBytes(StandardCharsets.UTF_8));
+            ctx.handleSimpleQuery(query, channel);
+        } finally {
+            query.release();
+        }
+
+        readReadyForQueryMessage(channel);
+        assertThat(channel.outboundMessages().size() , is(0));
+    }
```

Comment:
pls add a test expecting 1 query to fail

Reply:
In case you didn't see it, I added the test with https://github.com/crate/crate/pull/7252/commits/c4d79a43806635b47764dd7461671f25ba07cd64